File: src/brs/IndirectIncoming.java
package brs;

public class IndirectIncoming {
  private final long accountId;
  private final long transactionId;
  private final long amount;
  private final long quantity;
  private final int height;

  public IndirectIncoming(long accountId, long transactionId, long amount, long quantity, int height) {
      this.accountId = accountId;
      this.transactionId = transactionId;
      this.amount = amount;
      this.quantity = quantity;
      this.height = height;
  }

  public long getAccountId() {
      return accountId;
  }

  public long getTransactionId() {
      return transactionId;
  }

  public long getAmount() {
      return amount;
  }
  
  public long getQuantity() {
    return quantity;
  }

  public int getHeight() {
      return height;
  }

  @Override
  public boolean equals(Object o) {
      if (this == o) return true;
      if (o == null || getClass() != o.getClass()) return false;

      IndirectIncoming that = (IndirectIncoming) o;

      return accountId == that.accountId && transactionId == that.transactionId
          && amount == that.amount && quantity == that.quantity
          && height == that.height;
  }

  @Override
  public int hashCode() {
      int result = (int) (accountId ^ (accountId >>> 32));
      result = 31 * result + (int) (transactionId ^ (transactionId >>> 32));
      result = 31 * result + height + (int) amount + (int) quantity;
      return result;
  }
}

File: src/main/kotlin/brs/entity/IndirectIncoming.kt
package brs.entity

class IndirectIncoming(val accountId: Long, val transactionId: Long, val height: Int) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val that = other as IndirectIncoming?

        if (accountId != that!!.accountId) return false
        return if (transactionId != that.transactionId) false else height == that.height
    }

    override fun hashCode(): Int {
        var result = (accountId xor accountId.ushr(32)).toInt()
        result = 31 * result + (transactionId xor transactionId.ushr(32)).toInt()
        result = 31 * result + height
        return result
    }
}
